backend/internal/model/game_test.go

```
@@ -4,52 +4,55 @@ import (
+			game.ResetScore()
 		})
-		if tt.args.score.WhiteScore != 0 || tt.args.score.BlueScore != 0 {
-			t.Errorf("Score did not reset. Goals white: %v, Goals blue: %v", tt.args.score.WhiteScore, tt.args.score.BlueScore)
+		if tt.expectedWhiteScore != 0 || tt.expectedBlueScore != 0 {
+			t.Errorf("Score did not reset. Goals white: %v, Goals blue: %v", tt.expectedWhiteScore, tt.expectedBlueScore)
 		}
 	}
 }
 
 func TestAddGoal(t *testing.T) {
 	game := &Game{gameScore{3, 1}}
+
 	type args struct {
-		team int
+		name               string
+		teamID             int
+		expectedBlueScore  int
+		expectedWhiteScore int
+		expectedError      error
 	}
-	tests := []struct {
-		name             string
-		args             args
-		wantedBlueScore  int
-		wantedWhiteScore int
-	}{
-		{"should increment team white score by one", args{1}, 0, 1},
-		{"should increment team blue score by one", args{2}, 1, 0},
-		{"should not cause an error", args{-1}, 0, 0},
+	tests := []args{
+		{name: "should increment team white score by one", teamID: 1, expectedBlueScore: 0, expectedWhiteScore: 1, expectedError: nil},
+		{name: "should increment team blue score by one", teamID: 2, expectedBlueScore: 1, expectedWhiteScore: 0, expectedError: nil},
+		{name: "should cause an error when invalid team ID", teamID: -1, expectedBlueScore: 0, expectedWhiteScore: 0, expectedError: errors.New("bad team ID")},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			game.score.WhiteScore = 0
 			game.score.BlueScore = 0
-			game.AddGoal(tt.args.team)
-			assert.Equal(t, game.score.BlueScore, tt.wantedBlueScore, "blue team score changes incorrectly")
-			assert.Equal(t, game.score.WhiteScore, tt.wantedWhiteScore, "white team score changes incorrectly")
+			game.AddGoal(tt.teamID)
+			assert.Equal(t, game.score.BlueScore, tt.expectedBlueScore, "blue team score changes incorrectly")
+			assert.Equal(t, game.score.WhiteScore, tt.expectedWhiteScore, "white team score changes incorrectly")
```

errors are not checked

---

Changed